control/control_loop.go

```diff
@@ -38,7 +39,7 @@ type Loop struct {
 	activeBackgroundWorkers sync.WaitGroup
 	cancelCtx               context.Context
 	cancel                  context.CancelFunc
-	running                 bool
+	running                 atomic.Bool
```

Comment:
thoughts @JohnN193 ? this is what single encoder does for position

Reply:
this works!